sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -251,104 +252,31 @@ def _func_to_entrypoint(component_func, python_version='python3'):
   complete_component_code = dedecorated_component_src + '\n' + wrapper_code + '\n' + codegen.end()
   return complete_component_code
 
-class ImageBuilder(object):
+class ComponentBuilder(object):
   """ Component Builder. """
-  def __init__(self, gcs_base, target_image):
+  def __init__(self, gcs_staging, target_image):
     self._arc_docker_filename = 'dockerfile'
     self._arc_python_filename = 'main.py'
     self._arc_requirement_filename = 'requirements.txt'
-    self._tarball_filename = str(uuid.uuid4()) + '.tar.gz'
-    self._gcs_base = gcs_base
-    if not self._check_gcs_path(self._gcs_base):
-      raise Exception('ImageBuild __init__ failure.')
-    self._gcs_path = os.path.join(self._gcs_base, self._tarball_filename)
+    self._container_builder = ContainerBuilder(gcs_staging)
     self._target_image = target_image
 
-  def _wrap_files_in_tarball(self, tarball_path, files={}):
-    """ _wrap_files_in_tarball creates a tarball for all the input files
-    with the filename configured as the key of files """
-    if not tarball_path.endswith('.tar.gz'):
-      raise ValueError('the tarball path should end with .tar.gz')
-    with tarfile.open(tarball_path, 'w:gz') as tarball:
-      for key, value in files.items():
-        tarball.add(value, arcname=key)
-
-  def _prepare_buildfiles(self, local_tarball_path, docker_filename, python_filename=None, requirement_filename=None):
+  def _prepare_files(self, local_dir, docker_filename, python_filename=None, requirement_filename=None):
```

Comment:
It seems to me that `python_filename` and `requirement_filename` are redundant.
It's responsibility of the caller to ensure that all needed files are in `local_dir`.
Another reason for this function not to handle these files is that they're used by the `Dockerfile` which is not analyzed or modified by this function. If this function renames the python file or requirements file, then the Dockerfile won't be able for find them.

My suggestion is to remove these python-specific parameters from this function and change the caller to copy those files to the context dir.